error-handler/helpers/resultHelper.ts

```
@@ -0,0 +1,29 @@
+import { Result } from "../types/result";
+
+export function onSuccess<T, E>(
+    result: Result<T, E>,
+    handler: (data: T) => void
+  ) {
+    if (result.state === 'success') {
```

Create an enum object for these (success, failure, loading) instead of checking against hardcoded strings.